### ISSUE TITLE
[Feature Request] Support PRAUC metric on GPU

### ISSUE THREAD

**Issue by contepablod** [catboost/catboost#3094]
While AUC and Logloss work on GPU, PRAUC is missing, forcing users back to CPU-only training, which is slow for large datasets.

      cb_kwargs = dict(
            iterations=self.iterations,
            depth=6,
            learning_rate=0.05,
            l2_leaf_reg=3.0,
            loss_function="Logloss",
            # PRAUC is CPU-only inside CatBoost; AUC is GPU-supported.
            # We compute AUPRC externally via sklearn, so this is cosmetic.
            eval_metric="AUC" if self.task_type_ == "GPU" else "PRAUC",
            class_weights=[1.0, fraud_weight],
            random_seed=self.random_state,
            task_type=self.task_type_,
            allow_writing_files=False,
            verbose=False,
        )
